macauff/matching.py

```diff
@@ -24,27 +27,103 @@ class CrossMatch():
     A class to cross-match two photometric catalogues with one another, producing
     a composite catalogue of merged sources.
 
-    The class takes three paths, the locations of the metadata files containing
-    all of the necessary parameters for the cross-match, and outputs a file
-    containing the appropriate columns of the datasets plus additional derived
-    parameters.
-
     Parameters
     ----------
-    joint_file_path : string
-        A path to the location of the file containing the cross-match metadata.
-    cat_a_file_path : string
-        A path to the location of the file containing the catalogue "a" specific
-        metadata.
-    cat_b_file_path : string
-        A path to the location of the file containing the catalogue "b" specific
-        metadata.
+    chunks_folder_path : string
+        A path to the location of the folder containing one subfolder per chunk.
+    use_memmap_files : boolean
+        When set to True, memory mapped files are used for several internal
+        arrays. Reduces memory consumption bottlenecks at the cost of increased
+        I/O contention.
+    resume_file_path : string, optional
+        A path to the location of the file containing resume information for the
+        cross match.
     '''
 
-    def __init__(self, joint_file_path, cat_a_file_path, cat_b_file_path):
+    def __init__(self, chunks_folder_path, use_memmap_files=False, resume_file_path=None,
+                 walltime=None, end_within='00:10:00', polling_rate=1):
         '''
         Initialisation function for cross-match class.
         '''
+        self.chunks_folder_path = chunks_folder_path
```

Comment:
Combined with the above comment, the default `__init__` state is that MPI is going to be used, but it would be great if init could check for installed MPI and/or maybe a `use_MPI` flag and either skip straight to the small serial-only loop, or run all of the MPI setup + call and monitor code.

Reply:
As mentioned above, I've added a `use_mpi` flag to the constructor to enable/disable use of MPI (enabled by default).